src/Error/BaseErrorHandler.php

```
@@ -148,7 +148,7 @@ public function handleError(
         ?int $line = null,
         ?array $context = null
     ): bool {
-        if (error_reporting() === 0) {
+        if (!(error_reporting() & $code)) {
```

```suggestion
        if (error_reporting() & $code === 0) {
```

Negation makes expression harder to understand.

---

Note that it's a bitwise operator `&`, not a logical operator `&&`.

---

I used the bitwise operator as suggested by PHP at:
https://www.php.net/manual/en/migration80.incompatible.php

---

Bitwise operation result is integer, so it is better to compare with integer.  `!$integer` is same as `$integer === 0`, but it makes more clear that value 0 is (un)wanted.

---

I would put () around the & operation then.

---

I just ran and debugged the code. It appears $code ($errno) is never 0 in PHP8:
https://www.php.net/manual/en/function.set-error-handler.php

I think we're stuck with the bitwise since this is backwards compatible with older PHP versions. We could however wrap the entire code block in "if (error_reporting() & $code) {" if we absolutely need to get rid of the negation.

FYI: The vendors Composer & PHPUnit also use equal bitwise operation with negation to handle this.

---

@garas I tried locally and there's a test case failure with your suggested change.

```
1) Cake\Test\TestCase\Error\ErrorHandlerTest::testErrorSuppressed
Failed asserting that a string is empty.

../cakephp/tests/TestCase/Error/ErrorHandlerTest.php:224
```

---

Sounds like we have existing coverage for this change then?

---

Guess you could say that :).

---

Did you put () around the bitwise operation?

---

No, I just tried exactly was garas has suggested.

---

I have tried paras' code with and without () around the bitwise operation. It still prints / logs the silenced errors.

I've been running the original code from my pull request in local development/staging since last Friday without any issues.